clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -176,6 +177,20 @@ static RetTy parseOptionalCIRKeyword(AsmParser &parser, EnumTy defaultValue) {
   return static_cast<RetTy>(index);
 }
 
+/// Parse an enum from the keyword, return failure if the keyword is not found.
+template <typename EnumTy, typename RetTy = EnumTy>
+static ParseResult parseCIRKeyword(AsmParser &parser, RetTy &result) {
```

Comment:
Neat, probably we can reuse that for other things in this file, can you create an issue so we can track it?